components/gitlab/gitlab.app.js

```diff
@@ -0,0 +1,113 @@
+const axios = require("axios");
+const parseLinkHeader = require('parse-link-header');
+const uuid = require("uuid");
+
+module.exports = {
+  type: "app",
+  app: "gitlab",
+  propDefinitions: {
+    projectId: {
+      type: "string",
+      label: "Project ID",
+      description: "The project ID, as displayed in the main project page",
+      async options({ page, prevContext }) {
+        let url;
+        let requestParams = this._makeRequestParams();  // Basic request parameters
```

Comment:
Consider renaming this to `requestConfig` (and maybe the method to `_makeRequestConfig`) or something more generic. Initially I thought it was the actual `params` expected by `axios`.

Reply:
Yeah, I agree, I had the same issue when telling them apart with the query string params. I see that [axios refers to them as `config`](https://github.com/axios/axios#request-method-aliases) so I'll stick to that.